### PR TITLE
Display Multiple Transit Fares if Present

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR gives the `trip-details` component the power to display multiple fares if multiple fares are passed. It touches `core-utils` to make this work and is NOT backwards compatible. The change is what is returned by the `calculateFares()` method. I'm not sure if this warrants a breaking change. 

Some stories are also updated to include more fare information.

Screen Shot 2021-10-05 at 4 22 56 PM

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]

**Review by philip-cline — APPROVED** [review]

File: packages/trip-details/src/styled.js [hunk 1]
```diff
@@ -44,7 +44,15 @@ export const TNCFareCompanies = styled.span`
 
 export const Timing = styled.span``;
 
-export const TransitFare = styled.span``;
+export const TransitFare = styled.details`
+  display: inline-block;
+
+  > span {
+    display: block;
+    padding-left: 2ch;
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/styled.js:52]: Some of the CSS changes cause a slight, barely visible mislalignment:

image

File: packages/trip-details/src/index.js [hunk 2]
```diff
@@ -9,9 +9,36 @@ import { MoneyBillAlt } from "@styled-icons/fa-solid/MoneyBillAlt";
 import * as S from "./styled";
 import TripDetail from "./trip-detail";
 
+const TransitFare = ({ fareKey, fareNameFallback, keyMap, transitFares }) => {
+  const currentFare = transitFares[fareKey];
+
+  return (
+    <span>
+      {keyMap[fareKey] || fareNameFallback || fareKey}:{" "}
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.js:17]: The naming fallback is clever.

File: packages/trip-details/src/index.js [hunk 3]
```diff
@@ -137,8 +195,14 @@ function TripDetails({
 TripDetails.propTypes = {
   /** Used for additional styling with styled components for example. */
   className: PropTypes.string,
+  /** Determines which transit fare should be displayed by default, should there be multiple transit fare types */
+  configDefaultFare: PropTypes.string,
   /** Itinerary that the user has selected to view, contains multiple legs */
   itinerary: coreUtils.types.itineraryType.isRequired,
+  /** Mapping between fare keys and human-readable names for them */
+  keyMap: PropTypes.shape({
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.js:203]: Rename to `fareNames` or `fareKeyNameMap` or something like that?

File: packages/trip-details/src/index.js [hunk 4]
```diff
@@ -137,8 +195,14 @@ function TripDetails({
 TripDetails.propTypes = {
   /** Used for additional styling with styled components for example. */
   className: PropTypes.string,
+  /** Determines which transit fare should be displayed by default, should there be multiple transit fare types */
+  configDefaultFare: PropTypes.string,
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.js:199]: Why not `defaultFareKey`?

**Review by binh-dam-ibigroup — APPROVED** [review]
Just two suggestions for renaming props, but looks good otherwise.

File: packages/trip-details/src/styled.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/trip-details/src/styled.js]: Good catch thanks! Looks like there's a `0.25ch` padding on the caret from somewhere else.

File: packages/trip-details/src/index.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/trip-details/src/index.js]: Renamed in 088cc5ff030b873cf8e3b93a64a5a84377ce7142

File: packages/trip-details/src/styled.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/trip-details/src/styled.js]: Fixed in 088cc5ff030b873cf8e3b93a64a5a84377ce7142

**Pull request merged**